### DISCUSSION TITLE
How to get state variable from "A" slice in "B" slice?

### DISCUSSION THREAD

**Discussion by popstar7** [Finalgoal231/finalGoal#54] [Discuss about technical stack] [answered]
Hi, everyone
Thank you for your attention.
Now I want to build CRUD system in React-Redux-Toolkit.
But to get and modify fetched data in slice, I need to get data from another slice.
At first, I am going to use useSelector hook in slice.
But this is not correct way.
If you have any experience, please help me and share your solution

**Comment by Luis96920** [comment]
You're right that using the useSelector hook within a slice file is not the correct approach. Redux slices are supposed to be isolated and shouldn't depend directly on React-specific hooks like useSelector. However, there are proper ways to share and access data across slices. 

Here are very simple and common way.
please check:
You can create selectors in your slices to access specific parts of the state. Then, you can use those selectors in your components or other parts of your application where needed.
Example:

// sliceA.js
const sliceA = createSlice({
  name: 'sliceA',
  initialState: { data: [] },
  reducers: {
    setData: (state, action) => {
      state.data = action.payload;
    },
  },
});

export const selectSliceAData = (state) => state.sliceA.data;
export default sliceA.reducer;


If you need to access data from another slice during an asynchronous operation, you can use getState within a thunk to fetch the data from the store.
// sliceB.js
const sliceB = createSlice({
  name: 'sliceB',
  initialState: { items: [] },
  reducers: {
    setItems: (state, action) => {
      state.items = action.payload;
    },
  },
});

export const fetchAndModifyData = () => (dispatch, getState) => {
  const dataFromSliceA = getState().sliceA.data;

  // Perform operations using dataFromSliceA and then dispatch actions in sliceB
  dispatch(sliceB.actions.setItems(modifiedItems));
};

export default sliceB.reducer;

**Reply by noahstoner** [reply]
In A slice, export initialstate and in B, import it
But I can't confirm, it can be updated value.

**Comment by noahstoner** [comment, marked as answer]
We can think slice A as a module.
So we can export slice A and in slice B, we can import slice A.
with:
import sliceA from '../sliceB/sliceB.js"
